core/templates/pages/exploration-player-page/services/content-translation-manager.service.ts

```diff
@@ -0,0 +1,218 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to manage the content translations displayed.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { EventEmitter, Injectable } from '@angular/core';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { PlayerTranscriptService } from
+  'pages/exploration-player-page/services/player-transcript.service';
+import { StateCard } from
+  'domain/state_card/StateCardObjectFactory';
+import { WrittenTranslation } from
+  'domain/exploration/WrittenTranslationObjectFactory';
+import { SubtitledHtml } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { Schema } from 'services/schema-default-value.service';
+import { SchemaConstants } from
+  'components/forms/schema-based-editors/schema-constants';
+import INTERACTION_SPECS from 'pages/interaction-specs.constants.ajs';
+import { WrittenTranslations } from
+  'domain/exploration/WrittenTranslationsObjectFactory';
+import { SubtitledUnicode } from
+  'domain/exploration/SubtitledUnicodeObjectFactory';
+import { ExtensionTagAssemblerService } from
+  'services/extension-tag-assembler.service';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContentTranslationManagerService {
+  private explorationLanguageCode: string;
+  private onStateCardContentUpdateEmitter: EventEmitter<void> = (
+    new EventEmitter());
+  // The 'originalTranscript' is a copy of the transcript in the exploration
+  // language in it's initial state.
+  private originalTranscript: StateCard[] = [];
+
+  constructor(
+    private playerTranscriptService: PlayerTranscriptService,
+    private extensionTagAssemblerService: ExtensionTagAssemblerService
+  ) {}
+
+  init(explorationLanguageCode: string) : void {
+    this.explorationLanguageCode = explorationLanguageCode;
+    this.originalTranscript = cloneDeep(
+      this.playerTranscriptService.transcript);
+  }
+
+  get onStateCardContentUpdate(): EventEmitter<void> {
+    return this.onStateCardContentUpdateEmitter;
+  }
+
+  /**
+   * This method replaces the translatable content inside the player transcript
+   * service's StateCards. If the language code is set back to the original
+   * exploration language, the original transcript in the exploration language
+   * is restored, since it was previously modified from switching to another
+   * language previously. Note that learners can only freely switch content
```

Comment:
Slight rewording for clarity:

Note that learners can only freely switch content languages when they are on the first state and have not entered an answer yet (so, there are no response pairs), otherwise ...

(It took me a couple of false reads to realize that the "and" was still a part of the previous clause.)

Reply:
Done!